packages/task/src/browser/task-schema-updater.ts

```diff
@@ -532,13 +532,61 @@ const problemMatcher = {
     ]
 };
 
+const taskIdentifier: IJSONSchema = {
+    type: 'object',
+    additionalProperties: true,
+    properties: {
+        type: {
+            type: 'string',
+            description: 'The task identifier.'
+        }
+    }
+};
+
 const processTaskConfigurationSchema: IJSONSchema = {
     type: 'object',
     required: ['type', 'label', 'command'],
     properties: {
         label: taskLabel,
         type: defaultTaskType,
         ...commandAndArgs,
+        isBackground: {
+            type: 'boolean',
+            default: false,
+            description: 'Determines whether the task is a background task or not.'
+        },
+        dependsOn: {
+            anyOf: [
+                {
+                    type: 'string',
+                    description: 'Another task this task depends on.'
+                },
+                taskIdentifier,
+                {
+                    type: 'array',
+                    description: 'The other tasks this task depends on.',
+                    items: {
+                        anyOf: [
+                            {
+                                type: 'string'
+                            },
+                            taskIdentifier
+                        ]
+                    }
+                }
+            ],
+            description: 'Either a string representing another task or an array of other tasks that this task depends on.'
+        },
+        dependsOrder: {
+            type: 'string',
+            enum: ['parallel', 'sequence'],
+            enumDescriptions: [
+                'Run all dependsOn tasks in parallel.',
+                'Run all dependsOn tasks in sequence.'
+            ],
+            default: 'parallel',
+            description: 'Determines the order of the dependsOn tasks for this task. Note that this property is not recursive.'
+        },
```

Comment:
I assume the following is already covered through an existing CQ (contribution questionnaire).

Reply:
@vince-fugnitto 
Can you explain more about it ?
what is "CQ (contribution questionnaire"  ?

Reply:
Sure :) In general, whenever code is copied, we must register a CQ.
In this case we copied code from VS Code. More information is described in the repository's wiki.
I asked if its already covered since we have likely copied from this same file before when creating other task related PRs.

Reply:
@vince-fugnitto Was not it copied before already? If we had a CQ in the past it should not be necessary again.

Reply:
> I assume the following is already covered through an existing CQ (contribution questionnaire).

I think we would need to point to a specific existing CQ covering the same upstream file(s), where the copied code is identical (i.e. could be from same upstream version). Else we need a new CQ, covering the upstream file(s), for the VS Code version this was copied-from.

Reply:
Just to clarify, I have copied it from here : https://github.com/microsoft/vscode/blob/master/src/vs/workbench/contrib/tasks/common/jsonSchema_v2.ts

Reply:
I verified and could not easily find a CQ for which the file has previously been registered.
I guess the next best step is to be prudent and file for a CQ anyways.

I've created the CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21228

Reply:
Thanks @vince-fugnitto

Reply:
Just for information, the CQ has been successfully approved.